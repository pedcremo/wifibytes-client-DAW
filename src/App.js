import React, { Component } from 'react';
import { connect } from "react-redux";
import AppRoutes from './routes';
import { HashRouter } from 'react-router-dom';
import Navbar from './components/navbar';
import Footer from './components/footer';
import { getItems } from './actions/cartActions';

/**
 * @class
 * It defines the webpage structure.
 */
class App extends Component {
 /** render  
  * It renders the app view which loads the Navbar and Footer components and the AppRoutes component which has all the routes.
 */
 componentDidMount(){
    this.props.dispatch(getItems());
 }
 render() {
   return (
     <div className="App">
        <nav className="navbar navbar-expand-lg bg-white navbar-light">
            <Navbar cartItems={this.props.cartItems}/>
        </nav>
        <HashRouter>
            <div id="main" className="container-fluid pl-0 pr-0"> 
                <AppRoutes />
            </div> 
        </HashRouter>    
        <footer className="page-footer font-small bg-light pt-4">
            <Footer />
        </footer> 
     </div>
   );
 }
}

const mapStateToProps = state => ({
    cartItems: state.cartReducer
});

export default connect(mapStateToProps)(App);