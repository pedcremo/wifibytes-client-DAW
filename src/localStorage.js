const localStorage = store => next => action =>{
    if(action.localStorageSave){
        next(action)
        const currentStore = store.getState()
        console.log(currentStore)
        window.localStorage.setItem(action.reducer,JSON.stringify(currentStore[action.reducer]))
    }else if(action.localStorageUpdate){
        next(action)
        const currentStore = store.getState()
        window.localStorage.setItem(action.reducer,JSON.stringify(currentStore[action.reducer]))
    }else if(action.localStorageGet){
        if(window.localStorage.getItem(action.reducer)){
            const currentStore = JSON.parse(window.localStorage.getItem(action.reducer))
            store.dispatch({ type: action.action, currentStore });
        }
    }else
        next(action)
}

export default localStorage