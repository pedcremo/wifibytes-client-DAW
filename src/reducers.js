const initialState = {
    homeJSON: [],
    datosEmpresa:[],
    ratesJSON:[],
    tarifaJSON:[],
    activeRatesJSON:[]
  }
  â€‹
const rootReducer = (state = initialState, action) => state;

export default rootReducer;