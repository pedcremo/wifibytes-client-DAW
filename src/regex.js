/**
 * This is a library of regular expressions
 * We will use this regex to match for exemple input's values an validate them. 
 */
let Regex = {

    /**  
     * regexEmail check if a email is compose of ".", "@"
     */
    regexEmail: /^[_a-z0-9-]+(.[_a-z0-9-]+)*@[a-z0-9-]+(.[a-z0-9-]+)*(.[a-z]{2,4})$/,

    /**  
     * regexZip check if a zip has 5 digits
     */
    regexZip: /^(?:0[1-9]\d{3}|[1-4]\d{4}|5[0-2]\d{3})$/,

    /**  
     * regexPhone check if a phone has 9 digits and has started by 9 or 6 
     */
    regexPhone: /^[9|6]{1}([\d]{2}[-]*){3}[\d]{2}$/,
    /**
     * regexDni check if dni, nie or cif is valid
     */
    regexDni: /^\d{8}[a-zA-Z]{1}$/g,

    regexNie: /^[XxTtYyZz]{1}\d{7}[a-zA-Z]{1}$/g,

    regexCif: /^[a-zA-Z]{1}\d{7}[a-zA-Z0-9]{1}$/g,
    /**
     * regexFechNac check if fecha nacimiento is valid
     */
    regexFNAC: /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/g,

    regexSIM: /[0-9]{20}/,

    cvv: /^[0-9]{3,4}$/,
    cardOwner: /[a-zA-ZñÑáéíóúÁÉÍÓÚüÜ]{2,} [a-zA-ZñÑáéíóúÁÉÍÓÚüÜ]{2,}/g,
    iban: /([A-Z]{2})\s*\t*(\d\d)\s*\t*(\d\d\d\d)\s*\t*(\d\d\d\d)\s*\t*(\d\d)\s*\t*(\d\d\d\d\d\d\d\d\d\d)/g
}; //END Regex object


export {
    Regex
};
