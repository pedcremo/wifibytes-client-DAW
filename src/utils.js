/** @module Utils */

import {Settings} from "./settings";
import english from "./i18n/english.json";
import spanish from "./i18n/spanish.json";
import valencia from "./i18n/valencia.json";
import PaymentMethod from "../tests/json_endpoints/checkout_payment.json";

/**
 * Map to cache JSON already got from server.
 * TODO: Try to improve it. It is very basic at the moment
 * We catch every new json get from server and only expires when we close web tab or
 * we change user language
*/
let CACHE_TEMPLATES = new Map();
let userLanguage = english;


let Utils={

    /**  We set a language for user and store the same as a cookie afterwards we reload the page to apply changes
    * If there is no lang we set english as the default language for user
    * @param {string} lang - Choosen language.
    */
    setUserLanguage:function (lang=""){
        if (!lang) {
            let language=this.getCookie("language");
            if (!language || language=="") {
                var userLang = navigator.language;//Get navigator locales

                userLang=userLang.substring(0, 2).toUpperCase();

                switch (userLang) {
                    case "EN":
                        lang="english";
                        break;
                    case "ES":
                        lang="spanish";
                        break;
                    case "CA":
                        lang="valencia";
                        break;
                    default:
                        lang=Settings.defaultLanguage;
                }
                this.setCookie("language",lang,365);
                //lang=Settings.defaultLanguage;
            }else {
                lang=this.getCookie("language");
            }
        }else {
            if (this.getCookie("language")!==lang) {
                this.setCookie("language",lang,365);
                CACHE_TEMPLATES.clear(); //flush cache entries
                location.reload();//Reload current document
            }
        }
        switch (lang){
            case "spanish":
                userLanguage=spanish;
                break;
            case "valencia":
                userLanguage=valencia;
                break;
            default:
                userLanguage=english;
        }
    },
    /**
     * Post ajax call
     */
    post: function(url,data = {}){
        return new Promise((resolve, reject) => {
            let xhr = new XMLHttpRequest();
            xhr.open("POST", Settings.baseURL+url);
            xhr.setRequestHeader('Content-Type','application/json');
            xhr.onload = () => {
                if (xhr.status >= 200 && xhr.status < 300) {
                    resolve(xhr.response);
                } else {
                    reject(xhr.statusText);
                }
            };
            xhr.onerror = () => reject(xhr.statusText);
            xhr.send(JSON.stringify(data));
        })
    },
    /** Get is our ajax caller implemented as a promise
     * From Jake Archibald's Promises and Back:
     * http://www.html5rocks.com/en/tutorials/es6/promises/#toc-promisifying-xmlhttprequest
    */
    get: function (url,filterFunction=null) {
        /**Mocking /formaspago because is not ready in backend */
        /* if (url==='/formaspago')
            return filterFunction(PaymentMethod) */

        // Return a new promise.
        return new Promise(function(resolve, reject) {
            if (CACHE_TEMPLATES.has(url)) {
                    resolve(CACHE_TEMPLATES.get(url));
            }else{
                // Do the usual XHR stuff
                var req = new XMLHttpRequest();
                req.open("GET", Settings.baseURL+url);

                req.onload = function() {
                // This is called even on 404 etc
                // so check the status
                if (req.status == 200) {
                    // Resolve the promise with the response text
                    CACHE_TEMPLATES.set(url,JSON.parse(req.response));
                    if (filterFunction) {
                        let fFilter=filterFunction[0];
                        //resolve(fFilter(req.response));
                        resolve(fFilter(req.response,filterFunction[1]));
                    }else {
                        resolve(JSON.parse(req.response));
                    }
                }else {
                    // Otherwise reject with the status text
                    // which will hopefully be a meaningful error
                    reject(Error(req.statusText));
                }
                };

                // Handle network errors
                req.onerror = function() {
                reject(Error("Network Error"));
                };

                // Make the request
                req.send();
            }
        });
    },

    /** When we get a json array we only keep the ones that match user language
     * TODO: At the moment we only check that a 'lang' property from a gotten JSON
     * matchs current user language. Quite HARDWIRED.
    */
    filterPruneArrayByLang: function(jsonArray,langPropName){

        let lang =that.getUserLang();
        if (typeof jsonArray === "string" ) {
            jsonArray=JSON.parse(jsonArray);
        }
        let aux= jsonArray.filter((item) => {
            return item[langPropName] == lang;
        });
        return aux;
    },
    /** At the moment some endpoints on server side only have valencian and spanish content. And moreover it's
     * quite hardwired code to allow adding new languages. A really pain in the neck
     */
    getUserLang: function(){
        switch (this.getCookie("language")) {
            case "en":
                return "en";
            case "ca":
                return "va";
            case "es":
                return "es";
            default:
                return "es";
        }
    },
    /**
     * Get cookie by name
     * @param {string} cname
     */
    getCookie:function(cname) {
        try{
            var re = new RegExp(cname+"[\\s]*=[\\s]*([\\w.]*)","i");
            return document.cookie.match(re)[1];
        }catch(e){
            return "";
        }
    },
    /**
     * Create new cookie
     * @param {string} cname  Cookie name
     * @param {string} cvalue  Cookie value
     * @param {number} exdays Cookie expiration in days
     */
    setCookie:function(cname, cvalue, exdays) {
        var d = new Date();
        d.setTime(d.getTime() + (exdays*24*60*60*1000));
        var expires = "expires="+ d.toUTCString();
        document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/";
    },
    deleteCookie:function(name) {
        document.cookie = name +'=; Path=/; Expires=Thu, 01 Jan 1970 00:00:01 GMT;';
    },
    /**
     * Get key from userLanguage imported language selected taking into account user choosen lang
     * @param {string} key
     */
    translate:function(key) {
        if (userLanguage[key]) return userLanguage[key]
        else return key;
    },
    /**CSS Animation used in the app
     */
    randomAnimation() {
        const animations = ["bounce", "flash", "pulse", "rubberBand", "shake", "headShake", "swing", "tada", "wobble", "jello", "bounceIn", "bounceInDown", "bounceInLeft", "bounceInRight", "bounceInUp", "fadeIn", "fadeInDown", "fadeInDownBig", "fadeInLeft", "fadeInLeftBig", "fadeInRight", "fadeInRightBig", "fadeInUp", "fadeInUpBig", "rotateIn", "rotateInDownLeft", "rotateInDownRight", "rotateInUpLeft", "rotateInUpRight", "jackInTheBox", "rollIn", "zoomIn", "zoomInDown", "zoomInLeft", "zoomInRight", "zoomInUp", "slideInDown", "slideInLeft", "slideInRight", "slideInUp"]
        return "animated " + animations[Math.floor(Math.random() * (animations.length + 1))];
    },

    checkURL: function(hrefText) {
        if(hrefText){
            let reURL= /(http:\/\/|https:\/\/)/;
            if (hrefText.match(reURL)) {
                return hrefText;
            }else{
                return "http://"+hrefText;
            }
        }
    },
    deleteCookie:function(name) {
        document.cookie = name +'=; Path=/; Expires=Thu, 01 Jan 1970 00:00:01 GMT;';
    }
}; //END Utils object
let that = Utils;
export {Utils};
