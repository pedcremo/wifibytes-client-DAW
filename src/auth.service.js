import {Utils} from './utils';

let AuthService = {
    isAuth:function(){
        return new Promise(function(resolve, reject) {
            let token = Utils.getCookie("tokenAuth");
            if(token == "") reject(Error("No token"));
            Utils.post("/api-token-verify/",{'token':token}).then(function(res){
                res = JSON.parse(res);
                if(!Utils.getCookie("tokenAuth") || Utils.getCookie("tokenAuth") != res.token){
                    Utils.setCookie("tokenAuth",res.token,365)
                }
                resolve(res);
            }).catch((err)=>{
                reject(Error(err));
            });
        });
    },
    login:function(data){
        return new Promise(function(resolve, reject) {
            if(!data.username) reject("No username");
            if(!data.password) reject("No password");

            Utils.post("/api-token-auth/",data).then(function(res){
                console.log(res)
                res = JSON.parse(res);
                console.log(res)
                if(!Utils.getCookie("tokenAuth") || Utils.getCookie("tokenAuth") != res.token){
                    Utils.setCookie("tokenAuth",res.token,365)
                }
                resolve(res);
            }).catch((err)=>{
                reject(err);
            });
        });
    },
    register:function(data){
        return new Promise(function(resolve, reject) {
            if(!data.nombre) reject("No nombre");
            if(!data.apellido) reject("No apellido");
            if(!data.email) reject("No email");
            if(!data.cifnif) reject("No cifnif");
            if(!data.password) reject("No password");

            Utils.post("/cliente/",data).then(function(res){
                res = JSON.parse(res);
                if(!Utils.getCookie("tokenAuth") || Utils.getCookie("tokenAuth") != res.token){
                    Utils.setCookie("tokenAuth",res.token,365)
                }
                resolve(res);
            }).catch((err)=>{
                reject(err);
            });
        })
    },
    logout:function(){
        Utils.deleteCookie("tokenAuth")
    },
    recoverPass:function(diccionario){
        return new Promise(function(resolve, reject) {
            if(!diccionario.email) reject(Error("No email"));
            
            Utils.post("/clientenoreg/",diccionario)
            .then(function(res){
                resolve(res)
            })
            .catch((err)=>{
                reject(err)
            })
        });
    }
}
let that = AuthService;
export {AuthService};