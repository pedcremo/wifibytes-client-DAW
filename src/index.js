import React from 'react';
import ReactDOM from 'react-dom';
import {Provider} from 'react-redux';
import {store} from './store';
import I18n from 'redux-i18n';
import {Utils} from './utils';
import {translations} from './i18n/translations';

import App from './App';

/**
 * This is the entry point to our app. It defines a provider with the store and where is going to be displayed the app.
 *
 * <Provider /> is the higher-order component provided by React Redux that lets you bind Redux to React.
 *
 * Store holds the whole state tree of your application. The only way to change the state inside it is to dispatch an action on it.
 *
 * <App /> is a component which holds the webpage structure.
*/

document.addEventListener('DOMContentLoaded', function() {
  ReactDOM.render(
      <Provider store={store}>
        <I18n translations={translations} initialLang={Utils.getCookie('language')} fallbackLang={Utils.getUserLang()}>
          <App />
        </I18n>
      </Provider>
      , document.getElementById('root'));
});
