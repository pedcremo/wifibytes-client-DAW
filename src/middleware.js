import { Utils } from './utils';
import { REGISTER, LOGOUT, LOGIN, ASYNC_START, ASYNC_END, AUTH_SET, NOT_AUTH } from './constants/actionTypes';

const localStorage = (store) => (next) => (action) => {
	if (action.localStorageSave) {
		next(action);
		const currentStore = store.getState();
		window.localStorage.setItem(action.reducer, JSON.stringify(currentStore[action.reducer]));
	} else if (action.localStorageUpdate) {
		next(action);
		const currentStore = store.getState();
		window.localStorage.setItem(action.reducer, JSON.stringify(currentStore[action.reducer]));
	} else if (action.localStorageGet) {
		console.log(window.localStorage);
		if (window.localStorage.getItem(action.reducer)) {
			const currentStore = JSON.parse(window.localStorage.getItem(action.reducer));
			store.dispatch({ type: action.action, currentStore });
		}
	} else next(action);
};

const promiseMiddleware = store => next => action => {
  if (isPromise(action.payload)) {
      store.dispatch({ type: ASYNC_START, subtype: action.type });

      const currentView = store.getState().viewChangeCounter;
      const skipTracking = action.skipTracking;
      action.payload.then(
        res => {
          const currentState = store.getState()
          if (!skipTracking && currentState.viewChangeCounter !== currentView) {
            return
          }
          console.log('RESULT', res);
          action.payload = res;
          store.dispatch({ type: ASYNC_END, promise: action.payload });
          store.dispatch(action);
        },
        error => {
          const currentState = store.getState()
          if (!skipTracking && currentState.viewChangeCounter !== currentView) {
            return
          }
          console.log('ERROR', error);
          action.error = true;
          action.payload = {error : error};
          if (!action.skipTracking) {
            store.dispatch({ type: ASYNC_END, promise: action.payload });
          }
          store.dispatch(action);
        }
      );

      return;
    }

    next(action);
  };

const saveJWT = store => next => action => {
  //if (action.type === REGISTER || action.type === LOGIN) {
  if(action.type === LOGIN){
      if (!action.error) {
          Utils.setCookie('jwt',JSON.parse(action.payload).token ,365)
          window.localStorage.setItem('jwt', JSON.parse(action.payload).token);
          // agent.setToken(JSON.parse(action.payload).token);
      }
  } else if (action.type === LOGOUT) {
      Utils.setCookie('jwt','')
      window.localStorage.setItem('jwt', '');
      // agent.setToken(null);
  }

		return;
	}

	next(action);
};

const saveJWT = (store) => (next) => (action) => {
	if (action.type === REGISTER || action.type === LOGIN) {
		if (!action.error) {
			Utils.setCookie('jwt', action.payload.token, 365);
			window.localStorage.setItem('jwt', action.payload.token);
			// agent.setToken(JSON.parse(action.payload).token);
		}
	} else if (action.type === LOGOUT) {
		Utils.setCookie('jwt', '');
		window.localStorage.setItem('jwt', '');
		// agent.setToken(null);
	}

	next(action);
};

const isAuth = (store) => (next) => (action) => {
	if (action.isAuth) {
		let token = Utils.getCookie('jwt');
		if (token) {
			Utils.post('/api-token-verify/', { token: token }).then(
				(res) => {
					store.dispatch({
						type: AUTH_SET,
						user: res
					});
				},
				(error) => {
					console.log('ERROR isAuth Middleware : ', error);
					console.log(action);
					store.dispatch({
						type: NOT_AUTH
					});
				}
			);
		} else {
			next(action);
		}
	}
	next(action);
};

function isPromise(v) {
	return v && typeof v.then === 'function';
}
export { localStorage, saveJWT, promiseMiddleware, isAuth };
