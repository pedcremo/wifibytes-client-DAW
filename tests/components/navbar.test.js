import React from 'react';
import {Utils} from '../../src/utils';
import Navbar from '../../src/components/navbar';
import datosEmpresaJSON from '../json_endpoints/datos_empresa.json';
import ReactDOM from 'react-dom';
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() });

const $ = require('jquery');

jest.mock('../../src/utils');
const resp1 = datosEmpresaJSON;

Utils.get.mockResolvedValueOnce(resp1);
Utils.get.mockResolvedValue(resp1);
const nav = Enzyme.shallow(<Navbar />);

beforeEach(() => {
  // Set up our document body
  document.body.innerHTML =
      `<select id="langPicker" className="selectpicker" data-width="fit" value={this.state.value} onChange={this.handleLangPicker}>
          <option value='english'>English</option>
          <option value='spanish' >Español</option>
          <option value='valencia' >Valencià</option>
      </select>`;
});

it('Renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Navbar />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("We can check if Navbar component called the class constructor", () => {
  const wrapper = Enzyme.shallow(<Navbar />);
  expect(wrapper).toMatchSnapshot();
});

it('We check if data has been printed', () => {
  expect(nav.find('a[href="#/catalog"]').getElement().props.children).toHaveLength(3);
});

it('We can check if the handleLangPicker change', () => {
  nav.find('select').simulate('change', {target: {value:"english"}});
  expect(nav.render().find('select [selected]').val()).toEqual('english');
});