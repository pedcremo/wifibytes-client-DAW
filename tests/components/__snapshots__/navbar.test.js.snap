// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> dispatches event to show navbar data 1`] = `ShallowWrapper {}`;
